TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilderNodeListener.cs

```diff
@@ -0,0 +1,812 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.CodeModel;
+using TypeCobol.Compiler.Nodes;
+using TypeCobol.Compiler.Parser;
+
+namespace TypeCobol.Compiler.CupParser.NodeBuilder
+{
+    /// <summary>
+    /// A Node Listener that can also dispatch Program Class Builder actions
+    /// </summary>
+    public class ProgramClassBuilderNodeListener : NodeListener, IProgramClassBuilder
+    {
+        public virtual void OnNode(Node node, Program program)
+        {
+        }
+
+        public virtual void OnLevel1Definition(DataDefinition level1Node)
+        {
+
+        }
+
+        public virtual void Enter(Node node)
+        {
+        }
+
+        public virtual void Exit(Node node)
```

Comment:
Exit and OnNode method are called almost on the same lines from ProgramClassBuilder.
Is it useful to keep both ?

Reply:
Yes because sometime yes, and other time no.